ff_fat.c

```diff
@@ -1496,7 +1496,7 @@ uint32_t FF_CountFreeClusters( FF_IOManager_t * pxIOManager,
                     {
                         ulFreeClusters = FF_getLong( pxBuffer->pucBuffer, 488 );
 
-                        if( ulFreeClusters != ~0ul )
+                        if( ulFreeClusters != ( uint32_t ) ~0ul )
```

Comment:
I tend to avoid the use of the ambiguous 'ul' suffix and write the following:
~~~diff
-    if( ulFreeClusters != ( uint32_t ) ~0ul )
+    if( ulFreeClusters != ~( ( uint32_t ) 0U ) )
~~~

Reply:
the U suffix does not have any  effect now but I guess it is FreeRTOS style